common/lib/kconfig.ts

```diff
@@ -216,6 +217,76 @@ const ipfsGatewayUrl = (ipfs: string): string => `${ipfsGateway}${ipfsLinkToCid(
+    cid = uri.replace("https://api.gateway.ethswarm.org/bzz/", "");
+
+    // console.log("ipfsGetLink ~ uri res cid", uri, res, cid);
+  }
+  if (cid) {
+    // reconstruct ipfs uri
+    swarmLink = swarmCidToLink(cid);
+  }
+
+  return swarmLink;
+};
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// http uri : https://api.gateway.ethswarm.org/bzz/1fa18cf1aaee4727ecc266a86f1ef0f98b14771c7814d8cfb850a4b1c6d1359f
+// ipfs uri : swarm://1fa18cf1aaee4727ecc266a86f1ef0f98b14771c7814d8cfb850a4b1c6d1359f
+// => gateway url : https://api.gateway.ethswarm.org/bzz/1fa18cf1aaee4727ecc266a86f1ef0f98b14771c7814d8cfb850a4b1c6d1359f
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+const swarmToUrlHttp = (url: string): string => (url.startsWith("swarm://") ? swarmGatewayUrl(url) : url);
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////
+// cid : 1fa18cf1aaee4727ecc266a86f1ef0f98b14771c7814d8cfb850a4b1c6d1359f
+// => ipfs uri : swarm://1fa18cf1aaee4727ecc266a86f1ef0f98b14771c7814d8cfb850a4b1c6d1359f
+/////////////////////////////////////////////////////////////////////////////////////////////////////
+const swarmCidToLink = (cid: string): string => (cid ? `swarm://${cid}` : "");
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// ipfs uri : ipfs://bafkreieivwe2vhxx72iqbjibxabk5net4ah5lo3khekt6ojyn7cucek624
+// => cid : bafkreieivwe2vhxx72iqbjibxabk5net4ah5lo3khekt6ojyn7cucek624
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+const swarmLinkToCid = (swarm: string): string => swarm.replace(/^swarm:\/\//, "");
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Swarm reference : 1fa18cf1aaee4727ecc266a86f1ef0f98b14771c7814d8cfb850a4b1c6d1359f
+// => gateway url https://api.gateway.ethswarm.org/bzz/1fa18cf1aaee4727ecc266a86f1ef0f98b14771c7814d8cfb850a4b1c6d1359f
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+const swarmGatewayUrl = (swarm: string): string => `${swarmGateway}${swarmLinkToCid(swarm)}`;
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// cid from ipfs or swarm
+// => gataway url of ipfs or swarm
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+const cidGatewayUrl = (cid: string): string => (cid.startsWith("ipfs://") ? ipfsGatewayUrl(cid) : swarmGatewayUrl(cid));
```

Comment:
use **storage** instead of cid

const storageGatewayUrl = (link: string): string => 
link.startsWith("ipfs://") ? ipfsGatewayUrl(link) :
link.startsWith("swarm://") ? swarmGatewayUrl(link) : 
link

Reply:
done